### ISSUE TITLE
Taiko gets stuck when closing a modal and connecting back

### ISSUE THREAD

**Issue by varayal** [getgauge/taiko#550]
Taiko gets stuck when closing a modal and connecting back to the main page

### Steps to reproduce
1. Go to a page that opens a modal
2. Close the modal 
3. Find for an element on the main page

### Versions
```
 ↵ taiko --version                                                                    
Version: 0.8.0 (Chromium: 74.0.3723.0)
```

**Comment by NivedhaSenthil** [comment]
Can you please verify if it works fine with commit 3bb2ade ? pushed in a change which waits for reconnection when a tab is closed

**Comment by varayal** [comment]
the issue is now resolved on the latest taiko version, thank you very much for fixing this out!

**Comment by varayal** [comment]
please proceed to close this out

**Comment by varayal** [comment]
I just pull the latest version of taiko that resolves the Node id issue.

But now I'm getting this issue again. When the PayPal modal is closed it gets stuck again

Could you please check if this fix is merged in the lastest taiko version? 

`Version: 0.8.0 (Chromium: 74.0.3723.0) f6e741fcae1eb9b05a67ea825dad3bf7fe887f42`

**Comment by varayal** [comment]
sometimes I get this error on the console

`Error: Unexpected server response: 500
    at ClientRequest.req.on (/Users/victor.araya/projects/e2e-linio-gauge/node_modules/ws/lib/websocket.js:595:7)
    at ClientRequest.emit (events.js:189:13)
    at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:556:21)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:109:17)
    at Socket.socketOnData (_http_client.js:442:20)
    at Socket.emit (events.js:189:13)
    at addChunk (_stream_readable.js:284:12)
    at readableAddChunk (_stream_readable.js:265:11)
    at Socket.Readable.push (_stream_readable.js:220:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:94:17)
  Error Message: Connection closed [127.0.0.1:53811] cause: EOF
  Stacktrace:
`

**Comment by varayal** [comment]
I noticed if I added the below instruction 

`
await waitFor(intervalSecs(3));
`

the issue does not happen

but if I replace above instruction for a 

`
await text("¡GRACIAS, VICTOR!").exists(timeoutSecs(30))
`

Taiko cannot connect back to the browser
